corehq/apps/saved_reports/tasks.py

```diff
@@ -67,9 +100,10 @@ def initiate_queue_scheduled_reports():
 
 
 @serial_task('queue_scheduled_reports', queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
-def queue_scheduled_reports():
-    for report_id in create_records_for_scheduled_reports():
-        send_delayed_report(report_id)
+def queue_scheduled_reports(send_report_override_for_tests=None):
+    create_records_for_scheduled_reports()
```

Comment:
doesn't this function return a list of report IDs? Why is it necessary to call `get_queued_report_ids` straight after?

Ah, looks like the returned list could include ones marked as skipped. Would it not be worth changing it to return only queued ones rather than doing an extra query?

Reply:
The idea here is that if there are any reports that were previously queued but never processed, then this will fire kick off a task for those as well. This could possibly happen if the worker were killed and the celery task weren't requeued, or something like that. I don't expect it to happen often, but I wanted to add it as an extra layer of resilience. So `get_queued_report_ids` would pick up both newly queued and previously queued tasks.

In the processor, it checks again to make sure the task is queued, so this is unlikely to result in double sending even if there's a <15 minute lag in the queue and it ends up with repeats.